dynamic_programming/combination_sum_iv.py

```diff
@@ -0,0 +1,109 @@
+"""
+Question:
+You are given an array of distinct integers and you have to tell how many
+different ways of selecting the elements from the array are there such that
+the sum of chosen elements is equal to the target number tar.
+
+Example
+
+Input:
+N = 3
+target = 5
+array = [1, 2, 5]
+
+Output:
+9
+
+Approach:
+The basic idea is to go over recursively to find the way such that the sum
+of chosen elements is “tar”. For every element, we have two choices
+    1. Include the element in our set of chosen elements.
+    2. Don’t include the element in our set of chosen elements.
+"""
+
+
+from typing import List
```

Comment:
We target the code in this repo `3.10+` so this isn't necessary. 
You also use `list` in the code anyway
```suggestion
```